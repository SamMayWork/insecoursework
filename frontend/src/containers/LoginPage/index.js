/**
 * 
 * LoginPage
 * 
 * This is the first thing users of our App, at the '/' route
 */

import React from 'react';
import GoogleLogin from 'react-google-login';

import './loginPage.css';

const clientID = "817279853236-toe6rfq5ebg7rife6fvd82hh0eclpt3t.apps.googleusercontent.com";

function responseGoogle(response, history) {
  let token = response.uc.id_token;
  console.log(response, token);
  
  // SET USER SESSION DETAILS
  localStorage.setItem('real_name', response.profileObj.name);
  localStorage.setItem('email', response.profileObj.email);
  localStorage.setItem('token', token);
  
  fetch('/auth/test', {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Authorization': 'Bearer ' + token
    }
  }).then((res) => console.log(res)).then(
    (res) => {
    	history.push('/dashboard');
    }
  )
  .catch((error) => console.log(error));
  
  /*
  // AUTH TEST
  fetch('/forum/uac?register=true', {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Authorization': 'Bearer ' + token
    },
  }).then((res) => console.log(res)).then(
    (res) => {
    	history.push('/dashboard');
    }
  )
  .catch((error) => console.log(error));
  */
}

export function LoginPage(props) {
  return (
    <div className = "loginPage">
      <h1>
        UoP Forum
      </h1>
      <GoogleLogin
        buttonText="Login with Google"
        clientId={clientID}
        onSuccess={(response) => (responseGoogle(response, props.history))}
      />
    </div>
  );
}

export default LoginPage;
