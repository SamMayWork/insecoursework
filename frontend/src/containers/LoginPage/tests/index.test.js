import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import LoginPage from '../index';

const renderer = new ShallowRenderer();

describe('<LoginPage />', () => {
	describe('Validation tests', () => {
		it('should render and match the snapshot', () => {
		  renderer.render(<LoginPage />);
		  const renderedOutput = renderer.getRenderOutput();
		  expect(renderedOutput).toMatchSnapshot();
		});
	});
});
