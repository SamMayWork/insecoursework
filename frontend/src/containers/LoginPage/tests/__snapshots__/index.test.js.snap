// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> Validation tests should render and match the snapshot 1`] = `
<div
  className="loginPage"
>
  <h1>
    UoP Forum
  </h1>
  <u
    accessType="online"
    buttonText="Login with Google"
    clientId="817279853236-toe6rfq5ebg7rife6fvd82hh0eclpt3t.apps.googleusercontent.com"
    cookiePolicy="single_host_origin"
    disabledStyle={
      Object {
        "opacity": 0.6,
      }
    }
    fetchBasicProfile={true}
    icon={true}
    isSignedIn={false}
    jsSrc="https://apis.google.com/js/api.js"
    onRequest={[Function]}
    onSuccess={[Function]}
    prompt=""
    scope="profile email"
    tag="button"
    theme="light"
    type="button"
    uxMode="popup"
  />
</div>
`;
